File: b_rank_skillcheck_archive/name_card.py
# 実行時間 40ms
i = 0
n, m = list(map(int, input().split()))

while True:
    if n * 2 * i + 1 <= m and m <= n * 2 * (i + 1):
        l = []

        for j in range(1, n * 2 + 1):
            l.append(n * 2 * i + j)

        idx = l.index(m)

        l.reverse()
        print(l[idx])
        break
    i += 1

File: b_rank_skillcheck_archive/name_card.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int i = 0;
        int[] nm = ReadLine().Split().Select( int.Parse ).ToArray();

        while( true ) {
            if( nm[0] * 2 * i + 1 <= nm[1] & nm[1] <= nm[0] * 2 * ( i + 1 ) ) {
                var l = new System.Collections.Generic.List<int>( nm[0] * 2 );

                foreach( int j in Range( 1, nm[0] * 2 ) ) {
                    l.Add( nm[0] * 2 * i + j );
                }

                int idx = l.IndexOf( nm[1] );

                l.Reverse();
                WriteLine( l[idx] );
                break;
            }
            i++;
        }
        return;
    }
}
